plugins/timelib/timelib.cpp

```diff
@@ -568,33 +568,30 @@ TIMELIB_API void TIMELIB_CALL tlDateToString(size32_t &__lenResult, char* &__res
     __result = NULL;  // Return blank string on error
     __lenResult = 0;
 
-    // date is expected to be in form year*10000 + month * 100 + day, and must be later than 1900
-    // or we can't store it in a struct tm
```

Comment:
The behaviour when trying to store dates prior to 1900 in a struct tm is undefined. I think simply ignoring that fact leads to problems.

Reply:
(IIRC there were internal errors on OSX before this fix)

Reply:
While it is true that years prior to 1900 (which can be translated into a negative time_t value) are not explicitly supported by the standard, some systems do support it and my understanding was that Linux, Windows and macOS were all in that category.  I've never run into problems on macOS like this, but maybe I've just been lucky.  I'll research this further and do some more tests.  Also:  If it turns out that our supported platforms all allow negative time_t values then it would be worthwhile to add explicit unit tests for that condition.